plugins/routers/joint.routers.manhattan.js

```diff
@@ -382,7 +382,31 @@ joint.routers.manhattan = (function() {
             .difference(excludedEnds)
             // remove all elements whose type is listed in excludedTypes array
             .reject(function(element) {
-                return _.contains(opt.excludeTypes, element.get('type'));
+                // reject any element which is an ancestor of either source or target
+                var excludeAncestors = [];
```

Comment:
It's not necessary to compute the `excludeAncestors` array for every single element. All ancestors can be found in outer scope (source and target of the link stays the same). Note that performance matters in this case.